app/src/main/java/org/mozilla/focus/fragment/DownloadDialogFragment.kt

```diff
@@ -0,0 +1,128 @@
+import android.widget.TextView
+import mozilla.components.support.utils.DownloadUtils
+import org.mozilla.focus.R
+import org.mozilla.focus.telemetry.TelemetryWrapper
+import org.mozilla.focus.web.Download
+
+/**
+ * Fragment displaying a download dialog
+ */
+class DownloadDialogFragment : DialogFragment() {
+
+    override fun onCreateDialog(bundle: Bundle?): AlertDialog {
+        val fileName = arguments!!.getString("fileName")
+        val pendingDownload = arguments!!.getParcelable<Download>("download")
+
+        val builder = AlertDialog.Builder(context!!, R.style.DialogStyle)
+        builder.setCancelable(true)
+        builder.setTitle(getString(R.string.download_dialog_title))
+
+        val inflater = activity!!.layoutInflater
+        val dialogView = inflater.inflate(R.layout.download_dialog, null)
+        builder.setView(dialogView)
+
+        val downloadDialogIcon =
+            dialogView.findViewById<View>(R.id.download_dialog_icon) as ImageView
+        val downloadDialogMessage =
+            dialogView.findViewById<View>(R.id.download_dialog_file_name) as TextView
+        val downloadDialogCancelButton =
+            dialogView.findViewById<View>(R.id.download_dialog_cancel) as Button
+        val downloadDialogDownloadButton =
+            dialogView.findViewById<View>(R.id.download_dialog_download) as Button
+        val downloadDialogWarningMessage =
+            dialogView.findViewById<View>(R.id.download_dialog_warning) as TextView
+
+        downloadDialogIcon.setImageResource(R.drawable.ic_download)
+        downloadDialogMessage.text = fileName
+        downloadDialogCancelButton.text = getString(R.string.download_dialog_action_cancel)
+        downloadDialogDownloadButton.text = getString(R.string.download_dialog_action_download)
+        downloadDialogWarningMessage.text =
+                getSpannedTextFromHtml(R.string.download_dialog_warning, R.string.app_name)
+
+        val alert = builder.create()
+
+        setButtonOnClickListener(downloadDialogCancelButton, pendingDownload, false)
+        setButtonOnClickListener(downloadDialogDownloadButton, pendingDownload, true)
+
+        return alert
+    }
+
+    private fun setButtonOnClickListener(
+        button: Button,
+        pendingDownload: Download?,
+        shouldDownload: Boolean
+    ) {
+        button.setOnClickListener {
+            sendDownloadDialogButtonClicked(pendingDownload, shouldDownload)
+            TelemetryWrapper.downloadDialogDownloadEvent(shouldDownload)
+            dismiss()
+        }
+    }
+
+    private fun sendDownloadDialogButtonClicked(download: Download?, shouldDownload: Boolean) {
+        val listener = targetFragment as DownloadDialogListener?
+        listener?.onFinishDownloadDialog(download, shouldDownload)
+        dismiss()
+    }
+
+    @Suppress("DEPRECATION")
+    private fun getSpannedTextFromHtml(text: Int, replaceString: Int): Spanned {
+        return if (Build.VERSION.SDK_INT >= 24) {
```

Comment:
\[detekt\] INFO: \[style/MagicNumber\] Report magic numbers\. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is\. It's better to declare such numbers as constants and give them a proper name\. By default, \-1, 0, 1, and 2 are not considered to be magic numbers\.